Sources/IBLinterKit/Rules/ReuseIdentifierRule.swift

```diff
@@ -0,0 +1,52 @@
+import IBDecodable
+
+extension Rules {
+    
+    struct ReuseIdentifierRule: Rule {
+        
+        static let identifier = "reuse_identifier"
+        static let description = "Check that ReuseIdentifier same as class name."
+        
+        init(context: Context) {}
+        
+        func validate(storyboard: StoryboardFile) -> [Violation] {
+            guard let scenes = storyboard.document.scenes else { return [] }
+            let views = scenes.compactMap { $0.viewController?.viewController.rootView }
+            return views.flatMap { validate(for: $0, file: storyboard) }
+        }
+        
+        func validate(xib: XibFile) -> [Violation] {
+            guard let views = xib.document.views else { return [] }
+            return views.flatMap { validate(for: $0.view, file: xib) }
+        }
+        
+        private func validate<T: InterfaceBuilderFile>(for view: ViewProtocol, file: T) -> [Violation] {
+            let violation: [Violation] = {
+                // Currently only supported TableViewCell and CollectionViewCell
+                // UICollectionReusableView will be support after IBDecodable support its view.
+                // MKAnnotationView is not yet supported by interfacebuilder.
+                var reusableCells = [IBReusable & ViewProtocol]()
+                if let tableView = view as? TableView, let cells = tableView.prototypeCells {
+                    reusableCells += cells
+                } else if let collectionView = view as? CollectionView, let cells = collectionView.cells {
+                    reusableCells += cells
+                } else if let cell = view as? IBReusable & ViewProtocol {
+                    reusableCells += [cell]
```

Comment:
very special case :)
ok because cells are not in `TableView` or `CollectionView` `subviews` it seems

Reply:
I estimate this else if case is for XIB that defined TableViewCell or CollectionVIewCell without parent of TableView and CollectionView.